packages/cfd/src/features/types/components.types.ts

```diff
@@ -0,0 +1,348 @@
+import { ReactElement } from 'react';
+import { FormikHelpers as FormikActions, FormikValues } from 'formik';
+import { TCoreStores } from '@deriv/stores/types';
+import { MARKET_TYPE } from 'Helpers/cfd-config';
+import { ResidenceList, DetailsOfEachMT5Loginid } from '@deriv/api-types';
+import { TCFDPasswordReset } from './containers.types';
+import { TCFDPlatform, TTokens, TTradingPlatformAvailableAccount, TAccountCategory } from './shared.types';
+
+type TOnSubmit = (
+    index: number,
+    value: TFormValues,
+    setSubmitting: (isSubmitting: boolean) => void,
+    is_dirty?: boolean
+) => void;
+
+// cfd-account-copy
+export type TCFDAccountCopy = {
+    text?: string;
+    className: string;
+};
+
+// cfd-personal-details-form
+export type TCFDPersonalDetailsFormProps = {
+    index: number;
+    form_error?: string;
+    is_loading: boolean;
+    onSubmit: TOnSubmit;
+    initial_values: TFormValues;
+    changeable_fields?: string[];
+    residence_list: ResidenceList;
+};
+
+export type TValidatePersonalDetailsParams = {
+    values: TFormValues;
+    residence_list: ResidenceList;
+    account_opening_reason: TAccountOpeningReasonList;
+};
+
+export type TFindDefaultValuesInResidenceList = (params: {
+    citizen_text: string;
+    tax_residence_text: string;
+    place_of_birth_text?: string;
+    residence_list: ResidenceList;
+}) => {
+    citizen?: ResidenceList[0];
+    tax_residence?: ResidenceList[0];
+    place_of_birth?: ResidenceList[0];
+};
```

Comment:
```suggestion
export type TFindDefaultValuesInResidenceList = (params: {
    citizen_text: string;
    tax_residence_text: string;
    place_of_birth_text?: string;
    residence_list: ResidenceList;
}) => {
    citizen?: ResidenceList[0];
    tax_residence?: ResidenceList[0];
    place_of_birth?: ResidenceList[0];
};
```
There is overlap in keys here, params and return object could share the same type